packages/hardhat-core/src/internal/util/download.ts

```diff
@@ -11,8 +13,23 @@ export async function download(
   const { pipeline } = await import("stream");
   const { default: fetch } = await import("node-fetch");
   const streamPipeline = util.promisify(pipeline);
+  const fetchOptions: FetchOptions = {
+    timeout: timeoutMillis,
+  };
 
-  const response = await fetch(url, { timeout: timeoutMillis });
+  if (process.env.HTTPS_PROXY !== undefined) {
+    fetchOptions.agent = new HttpsProxyAgent(process.env.HTTPS_PROXY);
```

Comment:
There was some unnecessary casting here. The properties of `process.env` have type `string | undefined`, so if you check for that in the `if`, typescript will know that the type is `string` inside the block.

Reply:
Thanks! Yeah, I was struggling a bit on the TypeScript learning curve, but I'm slowly migrating a project to TypeScript currently.